src/pipeline.ts

```diff
@@ -465,6 +465,7 @@ export class GitHubWorkflow extends PipelineBase {
         name: 'Synthesize',
         permissions: {
           contents: github.JobPermission.READ,
+          idToken: this.useGitHubActionRole ? github.JobPermission.WRITE : github.JobPermission.NONE,
```

Comment:
This addition on its own will not do anything, because the steps in this job do not configure aws credentials.

It's actually non-trivial to add aws credentials here because the API `this.stepsToConfigureAws()` requires a region and account. We don't really have that in `jobForBuildStep`.

Reply:
Hi, yes this change alone is not enough, but we could make it work by adding aws credentials step as a prebuild step to the synth in while creating GitHubAction Workflow in our pipeline stack.